inlang/source-code/versioned-interfaces/message-bundle/src/interface.ts

```diff
@@ -0,0 +1,115 @@
+import { LanguageTag as Locale } from "@inlang/language-tag"
+import { Type, type Static } from "@sinclair/typebox"
+
+export type Literal = Static<typeof Literal>
+export const Literal = Type.Object({
+	type: Type.Literal("literal"),
+	value: Type.String(),
+})
+
+/**
+ * A (text) element that is translatable and rendered to the UI.
+ */
+export type Text = Static<typeof Text>
+export const Text = Type.Object({
+	type: Type.Literal("text"),
+	value: Type.String(),
+})
+
+export type VariableReference = Static<typeof VariableReference>
+export const VariableReference = Type.Object({
+	type: Type.Literal("variable"),
+	name: Type.String(),
+})
+
+export type Option = Static<typeof Option>
+export const Option = Type.Object({
+	name: Type.String(),
+	value: Type.Union([Literal, VariableReference]),
+})
+
+export type FunctionAnnotation = Static<typeof FunctionAnnotation>
+export const FunctionAnnotation = Type.Object({
+	type: Type.Literal("function"),
+	name: Type.String(),
+	options: Type.Array(Option),
+})
+
+/**
+ * An expression is a reference to a variable or a function.
+ *
+ * Think of expressions as elements that are rendered to a
+ * text value during runtime.
+ */
+export type Expression = Static<typeof Expression>
+export const Expression = Type.Object({
+	type: Type.Literal("expression"),
+	arg: Type.Union([Literal, VariableReference]),
```

Comment:
please no abbreviations. what is `arg`? singular argument or plural arguments?

EDIT: i see that message format 2.0 uses `arg` okay... can stay